From: Ben Guthro <ben@guthro.net>
Date: Thu, 3 Nov 2011 11:06:56 -0400
Subject: x86/microcode: check proper return code.
Origin: https://git.kernel.org/cgit/linux/kernel/git/konrad/xen.git/commit?id=f6c958ff0d00ffbf1cdc8fcf2f2a82f06fbbb5f4

After pulling in this change from your tree, I found the following bug,
when checking an enum value, which should be considered before inclusion:

Signed-off-by: Konrad Rzeszutek Wilk <konrad.wilk@oracle.com>
---
 arch/x86/kernel/microcode_xen.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/kernel/microcode_xen.c b/arch/x86/kernel/microcode_xen.c
index 9d2a06b..6a73957 100644
--- a/arch/x86/kernel/microcode_xen.c
+++ b/arch/x86/kernel/microcode_xen.c
@@ -157,7 +157,7 @@ static enum ucode_state xen_request_microcode_user(int cpu,
 	ret = UCODE_OK;
 
 out:
-	if (ret == 0)
+	if (ret == UCODE_OK)
 		uci->mc = uc;
 	else
 		vfree(uc);
